spark/src/test/scala/org/apache/spark/sql/delta/ImplicitDMLCastingSuite.scala

```diff
@@ -110,11 +122,15 @@ class ImplicitDMLCastingSuite extends QueryTest
               "updateAndMergeCastingFollowsAnsiEnabledFlag" ->
                 DeltaSQLConf.UPDATE_AND_MERGE_CASTING_FOLLOWS_ANSI_ENABLED_FLAG.key,
               "ansiEnabledFlag" -> SQLConf.ANSI_ENABLED.key).asJava)
-      case Some(exception: SparkThrowable) if sqlConfig.ansiEnabled =>
-        // With ANSI enabled the overflows are caught before the write operation.
+      case Some(exception: NumberFormatException) =>
+        val sparkThrowable = exception.asInstanceOf[SparkThrowable]
+        assert(sparkThrowable.getErrorClass == "CAST_INVALID_INPUT")
+        assert(sparkThrowable.getMessageParameters.get("sourceType") == "\"STRING\"")
+      case Some(exception: ArithmeticException) if sqlConfig.ansiEnabled =>
+        val sparkThrowable = exception.asInstanceOf[SparkThrowable]
```

Comment:
The method tries to cover too many different types of exceptions. We don't really see what's the expected exception for each case - is the newly added "map" test case throwing a NumberFormatException, a DeltaArithmeticException. More annoying is the fact that this validation may not catch if we start throwing a different exception in the future.

The easiest to solve this may be to add the expected error class to the test configuration